packages/flutter/lib/src/foundation/binding.dart

```diff
@@ -361,6 +364,32 @@ abstract class BindingBase {
     return instance!;
   }
 
+  /// The type of the current binding, if any, or else null.
+  ///
+  /// This may be useful in asserts to verify that the binding has not been initialized
+  /// before the point in the application code that wants to initialize the binding, or
+  /// to verify that the binding is the one that is expected.
+  ///
+  /// For example, if an application uses [Zone]s to report uncaught execptions, it may
+  /// need to ensure that `ensureInitialized()` has not yet been invoked on any binding
+  /// at the point where it configures the zone and initializes the binding.
+  ///
+  /// If this returns null, the binding has not been initialized.
+  ///
+  /// If this returns a non-null value, in returns the type of the binding instance.
+  ///
+  /// To obtain the binding itself, consider the `instance` getter on the [BindingBase]
+  /// subclass or mixin.
+  ///
```

Comment:
Document that calling this is only valid in debug mode and will return null in non-debug mode even when the bindings are initialized?

Reply:
oh yikes, good call, i totally forgot to mention that